src/NServiceBus.Core.Tests/Sagas/InvokeSagaNotFoundBehaviorTests.cs

```diff
@@ -0,0 +1,100 @@
+﻿namespace NServiceBus.Core.Tests.Sagas
+{
+    using System;
+    using NServiceBus.Pipeline;
+    using NServiceBus.Sagas;
+    using NUnit.Framework;
+    using ObjectBuilder;
+    using Unicast.Messages;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+
+    [TestFixture]
+    public class InvokeSagaNotFoundBehaviorTests
+    {
+        [SetUp]
+        public void SetupTests()
+        {
+
+            message = new LogicalMessage(new MessageMetadata(typeof(TestMessage)), new TestMessage(), null);
+
+            behavior = new InvokeSagaNotFoundBehavior();
+
+            incomingContext = new IncomingLogicalMessageContext(
+                message,
+                "messageId",
+                "replyToAddress",
+                new Dictionary<string, string>(),
+                null);
+
+            builder = new FuncBuilder();
+
+            incomingContext.Set<IBuilder>(builder);
+        }
+
+        [Test]
+        public void SagaNotFound_handlers_are_not_called_when_saga_is_found()
+        {
+            var validSagaHandler = new HandleSagaNotFoundValid();
+
+            builder.Register<IHandleSagaNotFound>(() => new HandleSagaNotFoundInvalid1());
+            builder.Register<IHandleSagaNotFound>(() => validSagaHandler);
+
+            Assert.That(async () => await behavior.Invoke(incomingContext, () => TaskEx.CompletedTask), Throws.Nothing);
+
+            Assert.False(validSagaHandler.Handled);
+        }
+
+        [Test]
+        public void Throw_friendly_invalidoperationexceptions_when_any_ihandlesaganotfound_handler_returns_null()
+        {
+            builder.Register<IHandleSagaNotFound>(() => new HandleSagaNotFoundInvalid1());
+            builder.Register<IHandleSagaNotFound>(() => new HandleSagaNotFoundValid());
+
+            Assert.That(async () => await behavior.Invoke(incomingContext, SetSagaNotFound), ThrowsEx.ExceptionWithFriendlyMessage(typeof(HandleSagaNotFoundInvalid1), "Handle"));
+        }
+
+
+        static Task SetSagaNotFound(IIncomingLogicalMessageContext context)
+        {
+            context.Extensions.Get<SagaInvocationResult>().SagaNotFound();
+            return TaskEx.CompletedTask;
+        }
+
+        class TestMessage : IMessage
+        { }
+
+        class HandleSagaNotFoundInvalid1 : IHandleSagaNotFound
```

Comment:
Here you use `Invalid` to indicate that you are using an implementation which does something which is not allowed. In other tests you had names like `ReturnsNull`. I think alignment would be helpful